1.15 Update

DrkMatr1984 opened the pull request:
Had to disable peaking into beacons while invisible, as 1.15 no longer has a BeaconInventory class or a way to retrieve Beacon Inventory.

addstar commented:
Hi @DrkMatr1984, thanks for the PR.

> as 1.15 no longer has a BeaconInventory class or a way to retrieve Beacon Inventory.

Can you explain what you mean by this or what makes you think there's no way to retrieve beacon inventory in 1.15? As far as I know, it's still there. Maybe the implementation changed, or it needs to be used in a different way (maybe using `InventoryView` now?).

Maybe if you link to some references where others have discussed it or confirmed it's gone? Thanks :slightly_smiling_face:

DrkMatr1984 commented:
https://hub.spigotmc.org/javadocs/bukkit/org/bukkit/block/Beacon.html
The beacon block interface itself no longer has a getInventory() method is what I actually meant. I suppose there is a Beacon inventory type in InventoryTypes and also an extension of Inventory called Beacon inventory still
(https://hub.spigotmc.org/javadocs/spigot/org/bukkit/inventory/BeaconInventory.html)
But as far as I could tell the only two ways to access them is to either run a runnable checking if an invisible player has any inventory opened, see if it's an instance of BeaconInventory, and then cast BeaconInventory onto it or to check the bukkit inventory events and see if invisible players are opening inventories of the type BeaconInventory and do the same. The blockstate of a beacon block no longer provides easy access to the inventory so I guess I was just being lazy and instead of adding a new listener and checking every inventory l cut it out.
I can try to add a listener to support opening Beacon inventories though in a few days and fix this PR for you if you'd rather have them supported. Sorry for being a lazy programmer :P

DrkMatr1984 commented:
This should do the trick ;)

DrkMatr1984 commented:
Ok, It's ready for merge :) I've fixed a whole host of bugs and broken code and its working as it was before now. I noticed in this fork that BungeeChat is a hard dependency and also the effects have been completely removed, as well as all the old hooks removed and disabled. If you're ever interested in an updated version without the gutting, I have updated [VNP-Original Edition](https://github.com/BukkitCodersUnited/VanishNoPacket) also to 1.15 and have fixed the hooks.
